gobblin-cluster/src/test/java/org/apache/gobblin/cluster/suite/IntegrationJobCancelSuite.java

```diff
@@ -17,14 +17,22 @@
 
 package org.apache.gobblin.cluster.suite;
 
+import java.io.File;
+
 import org.junit.Assert;
 
+import com.google.common.io.Files;
 import com.typesafe.config.Config;
 
 
 public class IntegrationJobCancelSuite extends IntegrationBasicSuite {
   public static final String JOB_ID = "job_HelloWorldTestJob_1234";
-  public static final String TASK_STATE_FILE = "/tmp/IntegrationJobCancelSuite/taskState/_RUNNING";
+  public static final String TASK_STATE_FILE;
+  static {
+    File tmpDir = Files.createTempDir();
```

Comment:
Does this need to happen in static block? Can it happen for the instance of a class? Accessing file system from static block looks strange to me. Also TASK_STATE_FILE seems to be like a constant, but it changes on every application run.